server/api/dev.js

```diff
@@ -0,0 +1,214 @@
+/*
+  Easier to refresh entire dev entry instead of upsert/update the dates
+  ??? What is the removed column for in entries?
+*/
+
+const db = require('../db');
+const { createLogger } = require('../logger');
+
+const NOW = new Date();
+const VOTE_MAX = 5;
+const BACKGROUND_COLORS = ['#000', '#FFF', '#4b91e3'];
+
+const logger = createLogger('API/DEV');
+
+const getDevContest = (status) => {
+  const contestStatuses = ['prerelease', 'reset', 'submission', 'review', 'voting', 'results'];
+  const contestIndex = contestStatuses.indexOf(status);
+  const isAfter = (lastStatus) => contestIndex > contestStatuses.indexOf(lastStatus);
+
+  const yearEnd = NOW.getMonth() === 11; // getMonth is zero-based
+  const date = NOW.toISOString().split('T')[0];
+  const fiveYears = 1000 * 60 * 60 * 24 * 365 * 5;
+  const pastest = new Date(NOW.getTime() - fiveYears - 1).toISOString();
+  const past = new Date(NOW.getTime() - fiveYears).toISOString();
+  const future = new Date(NOW.getTime() + fiveYears).toISOString();
+  const futurest = new Date(NOW.getTime() + fiveYears + 1).toISOString();
```

Comment:
[date-fns](https://date-fns.org/docs/Getting-Started) is what I've been using as my date time library. It's a little more readable than this and we don't have to do calculations like `1000 * 60 * 60 * 24 * 365 * 5` (which could be a class CONST instead of defined in the function).

```javascript
const yearEnd = format(NOW, 'MM') === 12; // or format(NOW, 'MMM') === 'Dec'
const date = formatISO(NOW, { representation: 'date' });
const pastest = sub(NOW, { years: 5, months: 1 });
const past = sub(NOW, { years: 5 });
const future = add(NOW, { years: 5 });
const futurest = add(NOW, { years: 5, months: 1 });
```

Reply:
Man, the Postgres timestamp was killing me. Used what you said above, but I actually needed to add `.toISOString()` to get this to work. You're right though, looks way better.